### ISSUE TITLE
SolidJS integration doesn't work with solid-js@1.6.x

### ISSUE THREAD

**Issue by pilcrowOnPaper** [withastro/astro#6501]
### What version of `astro` are you using?

2.1.2

### Are you using an SSR adapter? If so, which one?

none (static)

### What package manager are you using?

pnpm

### What operating system are you using?

MacOS

### What browser are you using?

irre

### Describe the Bug

Using `solid-js@1.6.x` breaks app when used with `@astrojs/solid-js@2.1.0`. Using `solid-js@1.5.x` fixes the issue. `pnpm dev` and `pnpm build` error:

```
 error   r is not defined
  File:
    /Users/user/code/astro-vercel/node_modules/.pnpm/solid-js@1.6.13/node_modules/solid-js/web/dist/server.js:297:5
  Code:
    296 |     timeoutHandle = setTimeout(() => reject("renderToString timed out"), timeoutMs);
    > 297 |   });
          |     ^
      298 |   return Promise.race([renderToStream(code, options), timeout]).then(html => {
      299 |     clearTimeout(timeoutHandle);
      300 |     return html;
  Stacktrace:
ReferenceError: r is not defined
    at eval (/node_modules/.pnpm/solid-js@1.6.13/node_modules/solid-js/web/dist/server.js:297:5)
    at Module.createRoot (/node_modules/.pnpm/solid-js@1.6.13/node_modules/solid-js/dist/server.js:49:14)
    at Module.renderToString (/node_modules/.pnpm/solid-js@1.6.13/node_modules/solid-js/web/dist/server.js:294:36)
    at Object.renderToStaticMarkup (@astrojs/solid-js/server.js:16:38)
    at renderFrameworkComponent (/node_modules/.pnpm/astro@2.1.2/node_modules/astro/dist/runtime/server/render/component.js:178:66)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

### Link to Minimal Reproducible Example

https://github.com/pilcrowOnPaper/astro-vercel

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Odas0R** [comment]
Same happening here!

**Comment by YuzeTT** [comment]
I have the same problem

**Comment by YuzeTT** [comment]
I found the solution: https://github.com/ddiu8081/chatgpt-demo/issues/206
change in `package.json`
From: "solid-js": "1.6",
To: "solid-js": "1.6.12",

**Comment by pilcrowOnPaper** [comment]
@YuzeTT 

So this?

| version | works? |
| - | - |
| 1.6.0 ~ 1.6.11 | ❌ |
| 1.6.12 | ✔️ |
| 1.6.13 | ❌ |

**Comment by btakita** [comment]
Is this an issue with solid-js with astro?

**Comment by XOP** [comment]
Same issue, after astro and solid-js updates had this issue.  
Light downgrade to 1.6.12 helped.

This is what works in my case:  
```
"@astrojs/solid-js": "^2.1.0",
"astro": "2.1.0",
"solid-js": "1.6.12"
```

**Comment by btakita** [comment]
Looks like this issue was fixed as of solid-js v1.6.14

https://github.com/solidjs/solid/issues/1613

**Comment by eighty4** [comment]
Putting this sample from Solid-js in an Astro page component does not render beyond the loading state. Logging in the async fn does not get invoked in the browser or the server. Putting client:load on the component fixes this problem. Is SSR like this possible with React and just not implemented for Solid-js?
